File: srm186/GolfScore.java
import java.util.*;

public class GolfScore {
	
	public int tally(int[] parValues, String[] scoreSheet) {
		int res = 0;
		for (int i = 0; i < parValues.length; i++) {
			switch (scoreSheet[i]) {
			
			case "hole in one":  res += 1;
								 continue;
			case "triple bogey": res += (parValues[i] + 3);
								 continue;
			case "double bogey": res += (parValues[i] + 2);
								 continue;
			case "bogey":		 res += (parValues[i] + 1);
								 continue;
			case "par": 		 res += parValues[i];
								 continue;
			case "birdie":		 res += (parValues[i] - 1);
								 continue;
			case "eagle":		 res += (parValues[i] - 2);
								 continue;
			case "albatross":    res += (parValues[i] - 3);
								 continue;
			}
		}
	return res;
	}
}

File: srm186/golf_score.py
class GolfScore:
	def tally(self, parValues, scoreSheet):
		ts = {"triple bogey": 3, "double bogey": 2, "bogey": 1, "par": 0, 
			"birdie": -1, "eagle": -2, "albatross": -3}
		res = 0
		for i in range(len(parValues)):
			if scoreSheet[i] == "hole in one": res += 1
			else: res += (parValues[i] + ts[scoreSheet[i]])
		
		return res